blockchain/blockchain.go

```diff
@@ -100,12 +100,12 @@ type PersistedState struct {
 }
 
 func LoadOrNewBlockchain(db dbm.DB, genesisDoc *genesis.GenesisDoc,
-	logger *logging.Logger) (*blockchain, error) {
+	logger *logging.Logger) (MutableBlockchain, error) {
 
 	logger = logger.WithScope("LoadOrNewBlockchain")
 	logger.InfoMsg("Trying to load blockchain state from database",
 		"database_key", stateKey)
-	blockchain, err := LoadBlockchain(db)
+	blockchain, err := loadBlockchain(db)
```

Comment:
This is another issue where I am changed my own leaning (though open to debate on this) towards 'exported by default' rather than 'unexported by default'. In some languages it's common to hide as many functions as possible in service of encapsulation. While I agree that it's a good idea to limit what essentially becomes your 'public API' I think hiding (unexporting) a lot of functions can be an anti-pattern. For example using other libraries in a way perhaps the author didn't anticipate I have been bitten by needlessly unexported functions that are close relatives of ones that are exported. This feels like the case here.

It is unlikely than many people find our `blockchain` type that interesting, but as  a principle I'd rather export.

Reply:
User wants to create a blockchain whenever he runs application. It can be created (first time) or loaded (other times). User shouldn't be involved in how we load/create blockchain. 
I vote for more encapsulation! however i's not a big change. I can also revert it.

Reply:
Okay, I'll take your vote, let's unexport this and `newBlockchain`

Reply:
Thanks Silas.